src/commands/pickFuncProcess.ts

```diff
@@ -83,32 +84,41 @@ async function startFuncTask(funcTask: Task, actionContext: IActionContext): Pro
             }
         });
 
-        const timeoutError: Error = new Error(localize('failedToFindFuncHost', 'Failed to detect running Functions host within "{0}" seconds. You may want to adjust the "{1}" setting.', timeoutInSeconds, `${extensionPrefix}.${settingKey}`));
         setTimeout(() => { reject(timeoutError); }, timeoutInSeconds * 1000);
     });
     await vscode.tasks.executeTask(funcTask);
     return await waitForStartPromise;
 }
 
-async function getInnermostWindowsPid(pid: string): Promise<string> {
+async function getInnermostWindowsPid(pid: string, timeoutInSeconds: number, timeoutError: Error): Promise<string> {
     const moduleName: string = 'windows-process-tree';
     const windowsProcessTree: IWindowsProcessTree | undefined = await tryFetchNodeModule<IWindowsProcessTree>(moduleName);
     if (!windowsProcessTree) {
         throw new Error(localize('noWindowsProcessTree', 'Failed to find dependency "{0}".', moduleName));
     }
 
-    let psTree: IProcessTreeNode | undefined = await new Promise<IProcessTreeNode | undefined>((resolve: (p: IProcessTreeNode | undefined) => void): void => {
-        windowsProcessTree.getProcessTree(Number(pid), resolve);
-    });
+    const maxTime: number = Date.now() + timeoutInSeconds * 1000;
+    while (Date.now() < maxTime) {
+        let psTree: IProcessTreeNode | undefined = await new Promise<IProcessTreeNode | undefined>((resolve: (p: IProcessTreeNode | undefined) => void): void => {
```

Comment:
Should this be part of the `while` loop?  Doesn't seem like we'd have to grab the powershell process again after the delay.

Reply:
It's not grabbing the powershell process - it's grabbing the whole tree of parent/child processes

Reply:
Well, either way, should we be grabbing the whole tree over and over again per while loop?

Reply:
I'm not sure how else I would do it? Plus 'windows-process-tree' is a native windows module and is very fast

Reply:
Couldn't you just grab it prior to entering the `while (Date.now() < maxTime)` loop (along with the error checking)?  I mean, I guess if it's fast, it doesn't really matter, just didn't seem necessary.

Reply:
Oh I see what you mean - and the answer is no I can't. The whole point of this PR is that the child processes haven't started yet and don't exist in the tree yet. The tree is different each time.